TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/Intake.java

```diff
@@ -46,24 +50,30 @@ public void initialize(HardwareMap hardwareMap, Telemetry telemetry) {
     public void execute(SmartController operator) {
         pidController.PIDControl(TARGET_DEGREE, motorLeft.getCurrentPosition());
 
-        if (operator.isButtonLeftBumper() && operator.isButtonRightBumper()) {
-            openBothIntakes();
-        } else if (operator.isButtonLeftBumper()) {
-            openLeftIntake();
-        } else if (operator.isButtonRightBumper()) {
-            openRightIntake();
-        } else if (operator.isLeftTriggerPressed() && operator.isRightTriggerPressed()) {
-            closeBothIntakes(operator);
-        } else if (operator.isLeftTriggerPressed()) {
-            handleLeftTrigger(operator);
-        } else if (operator.isRightTriggerPressed()) {
-            handleRightTrigger(operator);
-        } else {
-            stop();
-        }
+        ButtonUtils.whileHeld(operator.isButtonLeftBumper())
```

Comment:
This architecture looks awsome! I would like to suggest if we could rename some functions, for example:

ButtonUtils -> "ButtonListener"
whileHeld -> "while"  or "for"
then -> "execute" or "run"

this way your code would look like 
`ButtonListener.while(operator.isButtonLeftBumper()).and(...).execute(this::openBothIntakes)`

or

`ButtonListener.for(operator.isButtonLeftBumper()).and(...).run(this::openBothIntakes)`

Reply:
I have doubts about this type of code, since we neither a "Command" class for commands nor a "Button" class for buttons (like FRC), this code would be more unreliable, since you can put anything inside parameters. So keep in mind that if we don't have these types we can put this suggestion to another time. I think now we have reasons for creating these two types for a type safety code

Reply:
-----------------------
update: I just suggested to add a documentation, so for now we won't need to create these types and now I think my suggestions are ok to be valid